### PR TITLE
Field: UI & Code consistency Title -> Display name 

### LINKED ISSUE #24337: Unify field title/name in the UI and code
With changes introduced by https://github.com/grafana/grafana/pull/24024 we gonna have a weird situation, when field identity is called `title`. From UI pov this might be confusing, so we decided to unify the naming:

- use title (i.e. filterByTitleMatcher, field.config.title) in the codebase
- use Name in the UI

### PR DESCRIPTION
**Pull request by torkelo**
Closes https://github.com/grafana/grafana/issues/24337
Trying to align the terminology we use in code & UI for field name.  

In the UI we mostly only mention name (in transforms & field matchers). Except in the field config where it's named Title. 

It's also confusing that the field byName matcher is matching on getFieldTitle. 

So in an attempt to improve this inconsistency & avoid unnecessary confusion this PR. 

* [x] Rename getFieldTitle to getFieldDisplayName
* [x] Field config title rename to "Display name" 
* [x] Rename getFrameDisplayTitle to getFrameDisplayName
* [x] Rename field.state (FieldState) title property to displayName
* [x] Rename FieldConfig title to displayName 

Screenshot from 2020-05-11 12-30-03

### CODE REVIEW

File: packages/grafana-data/src/field/fieldState.ts [hunk 1]
```diff
@@ -30,39 +30,39 @@ export function getFrameDisplayTitle(frame: DataFrame, index?: number) {
...
 /**
- * Get an appropriate display title. If the 'title' is set, use that
+ * Get an appropriate display name. If the 'title' is set, use that
  */
-function calculateFieldTitle(field: Field, frame?: DataFrame, allFrames?: DataFrame[]): string {
+function calculateFieldDisplayName(field: Field, frame?: DataFrame, allFrames?: DataFrame[]): string {
   const hasConfigTitle = field.config?.title && field.config?.title.length;
 
-  let title = hasConfigTitle ? field.config!.title! : field.name;
+  let displayName = hasConfigTitle ? field.config!.title! : field.name;
```
Comment (hugohaggmark) [inline packages/grafana-data/src/field/fieldState.ts:55]: just so I understand, renaming `field.config.title` to `field.config.displayName` would mean migrations right? IMO we should do this now and not later :)

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 2]
```diff
@@ -106,20 +106,20 @@ export function applyFieldOverrides(options: ApplyFieldOverrideOptions): DataFra
...
+          name: displayName, // Generally appropriate (may include the series name if useful)
           labels: formatLabels(field.labels!),
           label: field.labels,
         },
       };
 
       field.state = {
         ...field.state,
-        title: title,
+        displayName: displayName,
```
Comment (hugohaggmark) [inline packages/grafana-data/src/field/fieldOverrides.ts:122]: nit: maybe change to
```suggestion
        displayName,
```

File: packages/grafana-data/src/field/fieldState.ts [hunk 3]
```diff
@@ -30,39 +30,39 @@ export function getFrameDisplayTitle(frame: DataFrame, index?: number) {
...
-    return title;
+    return displayName;
   }
 
   // This is an ugly exception for time field
   // For time series we should normally treat time field with same name
   // But in case it has a join source we should handle it as normal field
   if (field.type === FieldType.time && !field.labels) {
-    return title ?? 'Time';
+    return displayName ?? 'Time';
```
Comment (hugohaggmark) [inline packages/grafana-data/src/field/fieldState.ts:65]: nit: Do we have a const for the magic string `'Time'`?

**Review by hugohaggmark — APPROVED** [review]
Great work keeping the code/ui consistent and works as expected. Just left some small comments!

**Comment by dprokop** [comment]
> I am not renaming FieldConfig.title property here as that would involve panel migrations. Should we do that as well for consistency?

I think yes @torkelo. Migration should be fairly easy :)

File: packages/grafana-data/src/field/fieldState.ts (hunk 3 above)
Comment (dprokop) [reply packages/grafana-data/src/field/fieldState.ts:65]: yeah, would be great

**Review by dprokop — APPROVED** [review]
Great @torkelo !

File: packages/grafana-data/src/field/fieldState.ts (hunk 3 above)
Comment (torkelo) [reply packages/grafana-data/src/field/fieldState.ts]: added new constant :)

File: packages/grafana-data/src/field/fieldState.ts (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-data/src/field/fieldState.ts:55]: yes lets do this now -- not widely used yet, and better if we start more consistent

**Review by ryantxu — APPROVED** [review]
I vote we do this for 7.0

File: packages/grafana-ui/src/components/SingleStatShared/SingleStatBaseOptions.ts [hunk 4]
```diff
@@ -202,6 +202,15 @@ export function sharedSingleStatMigrationHandler(panel: PanelModel<SingleStatBas
...
 
+  if (previousVersion < 7.1) {
+    // move title to displayName
+    const oldTitle = (panel.fieldConfig.defaults as any).title;
+    if (oldTitle !== undefined && oldTitle !== null) {
+      panel.fieldConfig.defaults.displayName = oldTitle;
+      delete (panel.fieldConfig.defaults as any).title;
+    }
+  }
+
```
Comment (dprokop) [inline packages/grafana-ui/src/components/SingleStatShared/SingleStatBaseOptions.ts:213]: Shouldn't we handle this in the dashboard migrator? What if someone created i.e. Table panel in beta, with stable it's gonna no longer work.

Comment (torkelo) [reply packages/grafana-ui/src/components/SingleStatShared/SingleStatBaseOptions.ts:213]: It’s going to be messy to do in dashboard migrator as that runs before any panel migration so would have to take all previous field config structures in mind

Comment (torkelo) [reply packages/grafana-ui/src/components/SingleStatShared/SingleStatBaseOptions.ts:213]: Why would it not work for users running beta to stable? This migration will run for them as well

Comment (dprokop) [reply packages/grafana-ui/src/components/SingleStatShared/SingleStatBaseOptions.ts:213]: yes, but they will have to switch to stat-ish viz first right? But nvm, it's still gonna work as we are not adding nor removing any fields, so all good. Thinks this is good to merge! :)

**Pull request merged**